File: src/Player.java
public class Player {
    public static final int STONE = 0;
    public static final int SCISSORS = 1;
    public static final int PAPER = 2;

    private String name_;

    private int winCount_ = 0;

    public Player(String name) {
        name_ = name;
    }

    public int showHand() {
        int hand = 0;
        double randomNum = Math.random() * 3;
        if (randomNum < 1) {
            hand = STONE;
        }
        else if (randomNum < 2) {
            hand = SCISSORS;
        } else if (randomNum < 3) {
            hand = PAPER;
        }
        return hand;
    }

    public void notifyResult(boolean result) {
        if ( result == true ) {
            winCount_ += 1;
        }
    }

    public String getName() {
        return name_;
    }

    public int getWinCount() {
        return winCount_;
    }
}

File: python-oop/src/janken/player.py
class Player:

    # ジャンケンの手を表す定数
    STONE = 0  # グー
    SCISSORS = 1  # チョキ
    PAPER = 2  # パー

    def __init__(self, name, tactics):
        self.name = name
        self.win_count = 0
        self.tactics = tactics

    # ジャンケンの手を出す。
    def show_hand(self):
        hand = self.tactics.read_tactics()
        return hand

    # 審判から勝敗を聞く。勝ったら、引数 result は true
    def notify_result(self,result):
        if (result):
            self.win_count += 1

    # 自分の勝った回数を答える。
    def get_win_count(self):
        return self.win_count

    # 自分の名前を答える。
    def get_name(self):
        return self.name

# 継承し子クラス作成
class Player_STONE(Player):
    def show_hand(self):
        return self.STONE
